java/amazon-kinesis-producer/src/main/java/com/amazonaws/services/kinesis/producer/IKinesisProducer.java

```diff
@@ -18,7 +18,11 @@
 
     int getOutstandingRecordsCount();
 
-    default long getOldestRecordTimeInSeconds() {
+    default long getOldestRecordTimeInMillis() {
+        throw new UnsupportedOperationException("This method is not supported in this IKinesisProducer type");
+    }
+
+    default long getTotalUserRecordTimeoutsCount() {
```

Comment:
I think we don't need to expose this now. Can we remove this? We can add this in next iteration if required.

Reply:
done